woocommerce/woocommerce
# [PoC] Unlocking core private APIs in woocommerce-blocks

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR is intended to be the starting point for discussion about unlocking core private APIs to be used within Woo.
This PR also showcases how to use the [DimensionsTool](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/dimensions-tool/index.js#L52) which is a private feature from core as the aspect ratio control in the context of the product-image block, just like the core/image block does.





Closes #47924



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure you have a block compatible theme installed like `Twenty Twenty-Four`
2. Create a new post
3. Add a `Product Collection` block and select any option
4. When the product images are shown, select one of them
5. Open the right side Inspector panel by clicking on the Settings button located to the right of the Publish button
<img width="161" alt="image" src="https://github.com/user-attachments/assets/9fb9a207-0448-45d2-9080-06138ae3bff1">

6. Within the block inspector, click the Block tab, and then the Settings tab
7. Under the Image size panel, the new Aspect Ratio control should be shown
<img width="300" alt="image" src="https://github.com/user-attachments/assets/0a8e1844-6a9f-4a22-b8e6-f3801cefc2eb">

8. Notice that the default selected option is `Square 1:1` like requested in the first AC of the issue #47924
9. Changing the aspect ratio of the image does affect the selected image
<img width="1027" alt="image" src="https://github.com/user-attachments/assets/a7c9d508-f68a-4943-bcab-9146b0161a00">

10. Due to the way the Product Collection behaves, any change to the aspect ratio of one image will be also applied to the rest of the images within the collection.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
